### PR TITLE
Kicking guns now has a chance to fire them

### PR DESCRIPTION
**Pull request by press-mark**
I was working on #28862 when I thought of this dumb idea. Can be used for emergent gameplay.
:cl:
 * tweak: Be careful about kicking guns around!

### CODE REVIEW

**Comment by bathosbathtime** [comment]
I don't understand the, though small, negative reaction. Who's kicking guns enough for a 5% chance to misfire is an issue? Maybe I'm just dumb but this change looks good to me.

**Comment by bathosbathtime** [comment]
I've talked to my tip top lads about this and they saw issue in that drop testing is a significant part of gun manufacturing so only the cheapest pieces of garbage guns would fire from being kicked. While they were willing to accept that NT cheaped out on the hardware, they do humbly request that eguns and similar electronic firearms be excluded if possible.

**Comment by press-mark** [comment]
> I've talked to my tip top lads about this and they saw issue in that drop testing is a significant part of gun manufacturing so only the cheapest pieces of garbage guns would fire from being kicked. While they were willing to accept that NT cheaped out on the hardware, they do humbly request that eguns and similar electronic firearms be excluded if possible.

Alright energy weapons don't fire when kicked but projectile ones still can. Are there any specific guns that should have higher / lower chances of firing?

**Comment by jknpj** [comment]
crawlers don't deserve even more annoying gimmicks.

**Comment by ShiftyRail** [comment]
I think it's one of the most basic requirements of a firearm. Trigger guards and all that.

I want to like it because it's fun but immersion-wise it's a strange choice.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -466,3 +467,12 @@
 
 /obj/item/weapon/gun/proc/bullet_hitting(var/obj/item/projectile/P,var/atom/atarget)
 	return
+
+/obj/item/weapon/gun/kick_act(mob/living/carbon/human/H)
+	. = ..()
+	if(prob(kick_fire_chance))
+		var/list/targets = list()
+		for(var/turf/t in view(6))
```
Comment (Kurfursten) [inline code/modules/projectiles/gun.dm]: oview is probably better for this so that it doesn't (very confusingly) shoot at its own tile

**Comment by Gazathor** [comment]
would clowns be able to reliably fire guns via kicking them due to this?

**Comment by Kurfursten** [comment]
I wouldn't say reliable because it's 5% and picks a random tile to target within 6 tiles of view. Generally this means it will shoot 1 in 5 kicks and probably shoot an empty tile. Because the gun moves with each kick, it would also be hard to corral it toward the person you want to shoot with it.

**Pull request merged**